server/middleware/authorisationMiddleware.ts

```diff
@@ -1,21 +1,30 @@
-import type { RequestHandler } from 'express'
+import { RequestHandler } from 'express'
+import createError from 'http-errors'
 import { AccessMode } from '../@types/Handover'
-import URLs from '../routes/URLs'
 
 export default function authorisationMiddleware(): RequestHandler {
   return (req, res, next) => {
     if (req.services.sessionService.getPrincipalDetails()) {
-      if (
-        req.services.sessionService.getPrincipalDetails().accessMode === AccessMode.READ_ONLY &&
-        req.originalUrl?.split('?')[0] !== URLs.PLAN_OVERVIEW
-      ) {
-        req.session.returnTo = req.originalUrl
-        return res.redirect('/sign-in')
-      }
       return next()
     }
 
     req.session.returnTo = req.originalUrl
     return res.redirect('/sign-in')
   }
 }
+
+export function hasAccessMode(requiredAccessMode: AccessMode): RequestHandler {
```

Comment:
Can we rename this to `validateAccessMode`?

Reply:
renamed in https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/206/commits/db72907e836bee0b129166ca960242c987f07028